### PR TITLE
870 feature request add dataclass for checkresults

### LINKED ISSUE #870: [✨ Feature request]: Add dataclass for CheckResults
### Description

Must feature ok vs not ok. utilization number (and Factor of Safety). can also have required and provided amounts.

Enhancement from what Enrique already made in other branch.

### Acceptance Criteria

- [x] Code
- [x] Tests

### Dependencies

_No response_

### Confirmation

- [x] I have checked that this feature does not already exist in the issue tracker.
- [x] I have read and understood the contribution guidelines.

### PR DESCRIPTION
**Pull request by GerjanDorgelo**
## Description

Dataclass for the storage of check results. features utilization as unity check and the factor of safety, which are the reverse.

Can be initizated in 5 ways:
- CheckResult.from_bool(is_ok=True)
          -> CheckResult with is_ok=True, other fields None
- CheckResult.from_unity_check(unity_check=0.8)
          -> CheckResult with unity_check=0.8, factor_of_safety=1.25, is_ok=True, other fields None
- CheckResult.from_factor_of_safety(factor_of_safety=1.25)
          -> CheckResult with factor_of_safety=1.25, unity_check=0.8, is_ok=True, other fields None
- CheckResult.from_comparison(provided=80, required=100, operator="<=")
          -> CheckResult with unity_check=0.8, factor_of_safety=1.25, is_ok=True

Or it cam be done from CheckResult(..) directly, but must included consistent combination of variables. For example:
  - CheckResult(is_ok=True, unity_check=0.8) is valid.
  - CheckResult(is_ok=False, unity_check=0.8) will raise a ValueError.

Please note: When either `provided` or `required` is zero, or the operator is "==" or "!=", special handling is applied
        to avoid division by zero and ensure meaningful results. Unity check and factor of safety calculations are adjusted such
        that they yield either 0.0 or infinity in these edge cases, reflecting pass/fail status appropriately.

Fixes #870

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: blueprints/checks/check_result.py [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+
+    provided: float | None = None
+    required: float | None = None
+    operator: str = "<="
+    unity_check: float | None = None
+    factor_of_safety: float | None = None
+    is_ok: bool | None = None
+
+    @classmethod
+    def from_comparison(cls, provided: float, required: float, operator: str = "<=") -> "CheckResult":
```
Comment (egarciamendez) [inline blueprints/checks/check_result.py]:
```suggestion
    def from_comparison(cls, provided: float, required: float, operator: str = Literal["<", "<=", "==", ">=", ">", "!="] = "<=") -> Self:
```

File: blueprints/checks/check_result.py [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+
+        Returns
+        -------
+        CheckResult
+            A new CheckResult instance with the specified values.
+        """
+        return cls(provided=provided, required=required, operator=operator)
+
+    @classmethod
+    def from_unity_check(cls, unity_check: float) -> "CheckResult":
```
Comment (egarciamendez) [inline blueprints/checks/check_result.py]:
```suggestion
    def from_unity_check(cls, unity_check: float) -> Self:
```

File: blueprints/checks/check_result.py [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+
+        Returns
+        -------
+        CheckResult
+            A new CheckResult instance with unity_check=unity_check.
+        """
+        return cls(unity_check=unity_check)
+
+    @classmethod
+    def from_factor_of_safety(cls, factor_of_safety: float) -> "CheckResult":
```
Comment (egarciamendez) [inline blueprints/checks/check_result.py]:
```suggestion
    def from_factor_of_safety(cls, factor_of_safety: float) -> Self:
```

**Review by egarciamendez — APPROVED** [review]
Some minor changes, but ready to 🚢 after that
